### PR TITLE
GORM migration pt3

### PR DESCRIPTION
**Pull request by peterdeme**
3 changes:

- Removed NewRecord
- Count() return value is int64
- Update became Updates

### CODE REVIEW

File: association.go [hunk 1]
```diff
@@ -258,9 +258,9 @@ func (association *Association) Clear() *Association {
 }
 
 // Count return the count of current associations
-func (association *Association) Count() int {
+func (association *Association) Count() int64 {
```
Comment (peterdeme) [inline association.go:261]: Change number 1
https://github.com/go-gorm/gorm/blob/02b7e26f6b5dcdc49797cc44c26a255a69f3aff3/finisher_api.go#L438

File: association_test.go [hunk 2]
```diff
@@ -781,7 +781,7 @@ func TestRelated(t *testing.T) {
 
 	user1 = User{}
 	DB.Model(&address1).Related(&user1, "BillingAddressId")
-	if DB.NewRecord(user1) {
+	if user1.Id == 0 {
```
Comment (peterdeme) [inline association_test.go:784]: Change number 2
https://github.com/go-gorm/gorm/issues/3400

File: join_table_test.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func (*PersonAddress) Add(handler gorm.JoinTableHandlerInterface, db *gorm.DB, f
 	if result := db.Unscoped().Model(&PersonAddress{}).Where(map[string]interface{}{
 		"person_id":  foreignPrimaryKey,
 		"address_id": associationPrimaryKey,
-	}).Update(map[string]interface{}{
+	}).Updates(map[string]interface{}{
```
Comment (peterdeme) [inline join_table_test.go:32]: Change number 3
https://github.com/go-gorm/gorm/blob/02b7e26f6b5dcdc49797cc44c26a255a69f3aff3/finisher_api.go#L404

**Review by jakubdal — APPROVED** [review]
The code looks ok, but I don't have enough context to judge if the PR itself makes sense but I trust you.

**Comment by peterdeme** [comment]
@jakubdal I will not betray you

**Pull request merged**